tests/Users/edit-user.spec.js

```diff
@@ -0,0 +1,127 @@
+
+    await expect(page.getByText("Edit user")).toBeVisible();
+    await expect(page.getByLabel("First name")).toHaveValue(users[0].firstName);
+  });
+
+  test("Reset button makes all values empty", async ({ page }) => {
+    const users = await getUsers(page);
+    await navigateToUserPage(page, users[0]);
+
+    await page.getByRole("button", { name: "Reset" }).click();
+
+    await expect(page.getByLabel("First name")).toBeEmpty();
+    await expect(page.getByLabel("Last name")).toBeEmpty();
+    await expect(page.getByLabel("Email")).toBeEmpty();
+    await expect(
+      page.getByLabel("Password", { exact: true })
+    ).not.toBeInViewport();
+    await expect(page.getByLabel("Phone", { exact: true })).toBeEmpty();
+    await expect(page.getByLabel("Phone2")).toBeEmpty();
+    await expect(page.getByLabel("Birth date")).toBeEmpty();
+    await expect(page.getByLabel("Note")).toBeEmpty();
+    await expect(page.getByLabel("Birth date")).toBeEmpty();
+    expect(page.locator("div").filter({ hasText: /^USER$/ })).toBeTruthy();
+  });
+
+  test("Submit button does not work if there are empty fields", async ({
+    page,
+  }) => {
+    const users = await getUsers(page);
+    await navigateToUserPage(page, users[users.length - 1]);
+
+    await page.getByRole("button", { name: "Reset" }).click();
+    await page.getByRole("button", { name: "Submit" }).click();
+
+    await expect(page.getByLabel("First name")).toBeEmpty();
+    await expect(page.getByLabel("Last name")).toBeEmpty();
+    await expect(page.getByLabel("Email")).toBeEmpty();
+    await expect(
+      page.locator("#input-23-messages").getByText("Input field is required")
+    ).toBeVisible();
+    await expect(
+      page.locator("#input-25-messages").getByText("Input field is required")
+    ).toBeVisible();
+    await expect(page.getByText("Email field is required")).toBeVisible();
+    await expect(
+      page.locator("#input-39-messages").getByText("Input field is required")
+    ).toBeVisible();
+  });
+
+  test("Edit user successfully", async ({ page }) => {
+    const users = await getUsers(page);
+    await navigateToUserPage(page, users[users.length - 1]);
+
+    const firstName = "firstName" + getRandomNumber();
+    const lastName = "lastName" + getRandomNumber();
+    const email = "email" + getRandomNumber() + "@gmail.com";
+
+    await page.getByRole("button", { name: "Reset" }).click();
+
+    await page.getByLabel("First name").fill(firstName);
+    await page.getByLabel("Last name").fill(lastName);
+    await page.getByLabel("Email").fill(email);
+    await page.getByRole("combobox").locator("i").nth(1).click();
+    await page.getByRole("option", { name: "USER" }).click();
+
+    await page.getByRole("button", { name: "Submit" }).click();
+    const response = await page.waitForResponse(response => response.request());
+    expect(response).toBeDefined(); 
+    expect(response.status()).toBe(200);
+    expect(response.ok).toBeTruthy();
```

Comment:
The lambda function passed to waitForResponse seems incorrect because response.request() returns the Request object associated with the response, not a boolean indicating whether or not to wait for this response.
Here is a suggestion on how to improve it:

// Wait for a response where the URL matches the pattern for the users endpoint with a UUID
const response = await page.waitForResponse(response => 
    /http:\/\/localhost:8080\/users\/[0-9a-fA-F-]+/.test(response.url()) && response.status() === 200
);
expect(response.ok()).toBeTruthy(); // note that we should use 'ok()'not just 'ok'.

Reply:
fixed